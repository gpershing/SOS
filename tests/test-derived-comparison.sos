//comparison of unnamed structs
x: bool = {1,2} == {1,2}
print(x) //True

//unnamed arrays
x = [1,2,3] == [1,2,3]
printb(x) //True

//derived types are passed by reference
//array
arr1: array int = [1,2,3]
arr2: array int = [1,2,3]
arr3: array int= [3,2,1]

out1: bool = arr1 == arr1
print(out1) //True

out2: bool =arr1 == arr2
print(out2) //False

out3: bool  = arr1 == arr3
printb(out3) //False


//struct
struct s = {int id}
s s1 = {1} 
s s2 = {1} 
s s3 = {2} 

out4: bool = s1 == s1
print(out4) //True

out5: bool  = s1 == s2
print(out5) //False

out6: bool  = s1 == s3
print(out6) //False
