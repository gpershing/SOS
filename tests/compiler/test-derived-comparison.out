True
True
True
False
False
True
False
False
